postCore/src/main/java/com/hamdan/forzenbook/post/core/viewmodel/BasePostViewModel.kt

```diff
@@ -84,14 +96,31 @@ abstract class BasePostViewModel(
 
     companion object {
         private const val POST_LENGTH_LIMIT = 256
-        private const val PLACEHOLDER_TOKEN = "qHIc17fdLgcL750SOpxajnA9aUzDRsTDPAZqJuBcwfkcXw7rfC0mCvXzh1yed0RD"
+        private const val PLACEHOLDER_TOKEN =
+            "9aH7Jr398ccXTy5ArWXdvOJN5FpWNgcoZNrY76VekmT7oApGI4Xp9hVRZtltrzba"
     }
 }
 
-fun BasePostViewModel.PostState.getText(): String {
-    return ((this as BasePostViewModel.PostState.Content).content as BasePostViewModel.PostContent.Text).text
+fun BasePostViewModel.PostState.asContentOrNull(): BasePostViewModel.PostContent? {
+    return try {
+        (this as BasePostViewModel.PostState.Content).content
+    } catch (e: Exception) {
+        null
+    }
 }
 
-fun BasePostViewModel.PostState.getUri(): String {
-    return ((this as BasePostViewModel.PostState.Content).content as BasePostViewModel.PostContent.Image).uri
+fun BasePostViewModel.PostState.asImageOrNull(): BasePostViewModel.PostContent.Image? {
+    return try {
+        (this as BasePostViewModel.PostState.Content).content as BasePostViewModel.PostContent.Image
+    } catch (e: Exception) {
+        null
+    }
+}
+
+fun BasePostViewModel.PostState.asTextOrNull(): BasePostViewModel.PostContent.Text? {
+    return try {
+        (this as BasePostViewModel.PostState.Content).content as BasePostViewModel.PostContent.Text
```

Comment:
As I was alluding to before, you needed safe `as` casting for these, `as?`

Reply:
ah, well, it looks like I reinvented the wheel, looked it up and understand the mistake xd, will fix.